### ISSUE TITLE
Feature Comparison with Vespa

### ISSUE THREAD

**Issue by connorbrinton** [jina-ai/jina#1848]
Hi there 🙂  I'm evaluating ML-powered search and ranking solutions for a small- to medium-size use case and am curious about the differences between Jina and [Vespa](https://vespa.ai/). Some shared attributes between the two solutions seem to include:
* Efficient vector search
* Inline vector generation using user-provided ML models
* Multi-node clustering

I haven't really been able to get a sense of why I should choose one product over the other though. Any thoughts on features that distinguish Jina from Vespa?

**Comment by FionnD** [comment]
Hey @connorbrinton , 

Really sorry for the delay in getting back to you, we just released 1.0 of Jina, and all hands were on deck! 

We have just taken a quick look at Vespa, and some of the differences we notice are:

- Jina is an end-to-end search system. You can use Jina to extract semantic embedding, store them, search them, and return them to a user. Vespa doesn't seem to cover the whole pipeline. For example, I can't see any front end interface like [Jina Box](https://github.com/jina-ai/jinabox.js/). 
- Both Vespa and Jina use distributed node architecture. With Jina, you don't just distribute replicas etc; you can distribute various aspects of your training pipeline on other machines. Significantly, you could add a GPU for all your embedding creation.  Maybe Vespa allows this, but I didn't dig deep enough into their code. 
- With Jina, you can use [Jina Hub ](https://hub.jina.ai/.) A community collection of pre-build custom apps. You can plug and play with some of the latest AI research here. 
- Jina also integrates with Hugging face directly. With just one line of code in your YAML file, you can pull any of their pre-trained models. This is hugely benetifally. The latest research from Facebook/Google is added daily to hugging face community system. I suspect you could manually set this up with Vespa. 
- As you mentioned, Vespa is more focused on user-provided ML models. Which Jina supports, but we also support community-integrated models. 
- In the area of multi-node clustering. I suspect Vespa is a little more advanced with node recovery. However this is something we are working on this quarter. 

In summary, I think it depends on your use case. If you want end-to-end support and/or the lastest ML models, I would suggest Jina. If you are more focused on node and distributed design, then Vespa might be a good option. 


I hope that answers your question a little bit? Happy to talk more in detail over [Slack](https://jina-ai.slack.com/ssb/redirect) if you are interested. 

Cheers! Fionn

**Comment by connorbrinton** [comment]
No worries about the delay, congrats on the release! 🙂 Thanks for summarizing those differences, this information is extremely helpful for evaluating Jina.

I actually just learned about a new SOTA model for sentence encoding by browsing Jina Hub's PRs :smile: I'll have to try it out!

Thanks again for the info!
